test_singledispatch.py

```diff
@@ -578,15 +581,192 @@ def _(arg):
 
         # Registering classes as callables doesn't work with annotations,
         # you need to pass the type explicitly.
-        @i.register(str)
-        class _:
+        @i.register(six.text_type)
+        class _(object):
```

Comment:
Why does addition of singledispatchmethod require changing these tests that presumably worked before?

Reply:
I reverted the change in aa76793, as it passes on Python 2.7 in its original form.

Reply:
I was bitten by old-style vs new-style classes while implementing `singledispatchmethod`, so I think I was just being overly cautious with this change